src/notebooks/controllers/installPythonControllerCommands.ts

```diff
@@ -22,6 +29,8 @@ import { IVSCodeNotebookController } from './types';
 export class InstallPythonControllerCommands implements IExtensionSingleActivationService {
     private showInstallPythonExtensionContext: ContextKey;
     private showInstallPythonContext: ContextKey;
+    // WeakSet of executing cells, so they get cleaned up on document close without worrying
+    private executingCells: WeakSet<NotebookCell> = new WeakSet<NotebookCell>();
```

Comment:
This feels like more work than it should be to track. But I believe this is currently what we need to do to get ExecutionState as it's not accessible directly from the Cell.

Reply:
Can we not track the notebook instead of tracking the cells?
i.e. `WeakSet<NotebookDocument>`. Then further down we don't need to iterate through all the cells to check if the notebook is running